src/libos/src/fs/procfs/mod.rs

```diff
@@ -135,13 +141,61 @@ impl DirProcINode for LockedProcRootINode {
             1 => Ok(String::from("..")),
             i => {
                 let file = self.0.read().unwrap();
-                if let Some(s) = file.non_volatile_entries.keys().nth(i - 2) {
-                    Ok(s.to_string())
+                if let Some(name) = file.non_volatile_entries.keys().nth(i - 2) {
+                    Ok(name.to_owned())
                 } else {
-                    // TODO: When to iterate the process table ?
-                    Err(FsError::EntryNotFound)
+                    let processes = get_all_processes();
+                    let prior_entries_len = 2 + file.non_volatile_entries.len();
+                    let process = processes
+                        .iter()
+                        .nth(i - prior_entries_len)
+                        .ok_or(FsError::EntryNotFound)?;
+                    Ok(process.pid().to_string())
                 }
             }
         }
     }
+
+    fn iterate_entries(&self, mut ctx: &mut DirentWriterContext) -> vfs::Result<usize> {
+        let file = self.0.read().unwrap();
+        let mut total_written_len = 0;
+        let idx = ctx.pos();
+
+        // Write first two special entries
+        if idx == 0 {
+            let this_inode = file.this.upgrade().unwrap();
+            write_inode_entry!(&mut ctx, ".", &this_inode, &mut total_written_len);
+        }
+        if idx <= 1 {
+            let parent_inode = file.parent.upgrade().unwrap();
```

Comment:
"let parent_inode = file.parent.upgrade().unwrap();" only used in this file. What is the reason?

Reply:
This is because the parent is the root inode itself,  since the inode itself exists, the pointer can be weak. Maybe we can just remove the parent member of root inode of procfs.